From 97155bfcf171ee0ad58c42a55672fdc6d91d7916 Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Mon, 19 Mar 2012 22:38:38 -0500
Subject: [PATCH 286/505] ENGR00177223 - EPDC fb: Add unit test documentation

- Initial check-in of new documentation format for EPDC unit test.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 all-suite.txt                            |  1 +
 test/doc/ut_um.toc                       |  4 ++
 test/mxc_fb_test/doc/FSL-UT-EPDC-001.txt | 94 ++++++++++++++++++++++++++++++++
 3 files changed, 99 insertions(+)
 create mode 100644 test/mxc_fb_test/doc/FSL-UT-EPDC-001.txt

diff --git a/all-suite.txt b/all-suite.txt
index a0e58a3..d7bea44 100644
--- a/all-suite.txt
+++ b/all-suite.txt
@@ -6,6 +6,7 @@
 # every night when we run autorun.sh.
 # Refer to linux-test/test-utils.sh for available <PLATFORM LISTS>
 #
+FSL-UT-EPDC-0001:autorun-epdc.sh:IMX6
 FSL-UT-PXP-0001:autorun-pxp.sh:IMX6
 FSL-UT-PXP-0002:pxp_v4l2_out_test.sh:
 FSL-UT-WDOG-0001:autorun-wdog.sh:IMX6
diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index a587f94..814d9f8 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -12,3 +12,7 @@ include::../pxp_v4l2_test/doc/FSL-UT-PXP-002.txt[]
 WDOG
 ----
 include::../wdog/doc/FSL-UT-WDOG-001.txt[]
+
+EPDC
+---
+include::../mxc_fb_test/doc/FSL-UT-EPDC-001.txt[]
diff --git a/test/mxc_fb_test/doc/FSL-UT-EPDC-001.txt b/test/mxc_fb_test/doc/FSL-UT-EPDC-001.txt
new file mode 100644
index 0000000..fae5add
--- /dev/null
+++ b/test/mxc_fb_test/doc/FSL-UT-EPDC-001.txt
@@ -0,0 +1,94 @@
+FSL-UT-EPDC-001
+~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+EPDC frame buffer test for display updates
+
+| Automated |
+YES (But manual observation is required to confirming correct updates on display panel)
+
+| Kernel Config Option |
+ CONFIG_FB=y
+ CONFIG_FB_MXC=y
+ CONFIG_FB_MXC_EINK_PANEL=y
+ CONFIG_MFD_MAX17135=y
+ CONFIG_REGULATOR_MAX17135=y
+ CONFIG_MXC_PXP=y
+ CONFIG_DMA_ENGINE=y
+ ...
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+ifdef::imx50[]
+On i.MX508 Armadillo2 board:
+- EPD daughter board must connet to PORT4 of the ARM2 base board.
+
+On i.MX508 EVK:
+- EPD daughter board must be connected to J14 connector or
+- E-Book add-on card must be connected to both J14 and J18 connectors
+endif::[]
+ifdef::imx6q[]
+N/A
+endif::[]
+
+| Test Procedure |
+EPDC fb test for display updates.
+
+.Features tested (no correlation to EPDC test numbers):
+* RGB565, Y8 framebuffer format
+* 0, 90, 180, 270 degree framebuffer rotation
+* Framebuffer panning
+* Use of the alternate framebuffer
+* Auto-waveform mode selection
+* Automatic update mode
+* The force-monochrome update feature and animation mode updates
+* Support for using a grayscale colormap
+* Snapshot, Queue, and Queue and Merge update schemes
+* The EPDC Collision Test mode
+
+.EPDC tests:
+. Basic Updates
+. Rotation Updates
+. Grayscale Framebuffer Updates
+. Auto-waveform Selection Updates
+. Panning Updates
+. Overlay Updates
+. Auto-Updates
+. Animation Mode Updates
+. Fast Updates
+. Partial to Full Update Transitions
+. Test Pixel Shifting Effect
+. Colormap Updates
+. Collision Test Mode
+. Stress Test
+
+ $ /unit_tests/mxc_epdc_fb_test.out
+ Usage: mxc_epdc_fb_test [-h] [-a] [-n]
+	-h Print this message
+	-a Enabled animation waveforms for fast updates (tests 8-9)
+	-p Provide a power down delay (in ms) for the EPDC driver
+	-u Select an update scheme
+		s - Snapshot update scheme
+		q - Queued update scheme
+		m - Queued update scheme with update merging
+	-n Execute the tests specified in expression
+	Expression is a set of comma-separated numeric ranges
+ Example:
+ /unit_tests/mxc_epdc_fb_test.out -n 1-3,5,7
+
+| Expected Result |
+The full set of tests should pass without any failure messages.
+
+For each test, a sequence of updates should be reflected on the screen.  For
+almost all tests, the text printed out in the debug console describes the image
+that should be observed on the screen.
+
+|====================================================================
+
+<<<
-- 
1.8.0

